pkg/common/utils/utils.go

```diff
@@ -230,14 +231,32 @@ func QueryVolumeDetailsUtil(ctx context.Context, m cnsvolume.Manager, volumeIds
 // If the datastore with dsURL can be found in the same datacenter as the given VC
 // and it is also found in the given datastoreList, return the reference of the datastore.
 // Otherwise, return error.
-func GetDatastoreRefByURLFromGivenDatastoreList(ctx context.Context, vc *cnsvsphere.VirtualCenter,
-	datastoreList []types.ManagedObjectReference, dsURL string) (*types.ManagedObjectReference, error) {
+func GetDatastoreRefByURLFromGivenDatastoreList(ctx context.Context, clusterFlavor cnstypes.CnsClusterFlavor,
```

Comment:
Changes to this method affects CreateVolume calls, unit tests are failing. Can you take a look